.editorconfig

```diff
@@ -1,15 +1,34 @@
+# https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties
 root = true
 
 [*]
 charset = utf-8
-end_of_line = lf
-insert_final_newline = true
 indent_style = space
 indent_size = 4
+end_of_line = lf
 trim_trailing_whitespace = true
+insert_final_newline = true
+
+# Unofficial config, support depends on the application reading the file.
+# https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#supported-by-a-limited-number-of-editors
+# https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#ideas-for-domain-specific-properties
+max_line_length = off
+quote_type = single
+spaces_around_operators = true
+spaces_around_brackets = none
+
+# https://en.wikipedia.org/wiki/Indent_style
+indent_brace_style = Allman
 
 [*.md]
 trim_trailing_whitespace = false
 
 [*.{yml,yaml}]
+indent_style = space
 indent_size = 2
+
+[*.{blade.php,html}]
+max_line_length = 400
+
+[*.{js,vue}]
+max_line_length = 140
```

Comment:
Do we really want to enforce this?

Reply:
Maybe not necessarily enforce it, but it instructs tools like prettier when it should move a long line of code to the next line, and this seems to match up pretty well with our code.